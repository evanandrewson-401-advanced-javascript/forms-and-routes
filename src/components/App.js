import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Header from './Header';
import Footer from './Footer';
import HomePage from '../containers/HomePage';
import ListPage from '../containers/ListPage';
import DetailPage from '../containers/DetailPage';

class App extends Component {
  render() {
    return (
      <Router>
        <Header />
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route path="/list" component={ListPage} />
          <Route path="/detail/:id" component={DetailPage}/>
        </Switch>
        <Footer />
      </Router>
    );
  }
}

export default App;
