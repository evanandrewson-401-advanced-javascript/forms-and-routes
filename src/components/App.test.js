import App from './App';
import React from 'react';
import { shallow } from 'enzyme';

describe('App', () => {
  // eslint-disable-next-line no-undef
  global.fetch = jest.fn(() => Promise.resolve({}));
  it('renders Characters', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
