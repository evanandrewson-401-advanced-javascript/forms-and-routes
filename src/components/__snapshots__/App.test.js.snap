// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders Characters 1`] = `
<BrowserRouter>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/list"
    />
    <Route
      component={[Function]}
      path="/detail/:id"
    />
  </Switch>
  <Footer />
</BrowserRouter>
`;
